pyiron/sphinx/base.py

```diff
@@ -1515,6 +1553,94 @@ def write_spins_constraints(self, file_name="spins.in", cwd=None, spins_str=None
             s.logger.debug("No magnetic moments")
 
 
+class Group(dict):
+    def __init__(self, *args, **kw):
+        super(Group, self).__init__(*args, **kw)
+
+    def __setitem__(self, key, value):
+        if isinstance(value, dict):
+            value = Group(value)
+        super(Group, self).__setitem__(key, value)
+
+    def __setattr__(self, key, value):
+        if isinstance(value, dict):
+            value = Group(value)
+        super(Group, self).__setitem__(key, value)
+
+    __getattr__ = dict.get
+    # __setattr__ = dict.__setitem__
+    __delattr__ = dict.__delitem__
+
+    def set(self, name, content):
+        self[name] = content
+
+    def set_group(self, name, group):
+        self.set(name, group)
+
+    def set_flag(self, flag, val=True):
+        self.set(flag, val)
+
+    def set_parameter(self, parameter, val):
+        self.set(parameter, val)
+
+    def remove(self, name):
+        del self[name]
+
+    def remove_group(self, group):
+        self.remove(group)
+
+    def remove_flag(self, flag):
+        self.remove(flag)
+
+    def remove_parameter(self, parameter):
+        self.remove(parameter)
+
+    def to_sx_str(self, content="__self__", level=0):
+        line = ""
+        if content == "__self__":
+            content = self
+
+        for k, v in content.items():
+            k = k.split("___")[0]
+
+            if k == "eCut":
```

Comment:
Do you mean writing e.g. `EnCut = 340 / 13.606` in userparameters.sx instead? I'm happy with that but didn't want to mess with the `Input.write_file()` method invoked to write userparameters.

Reply:
No I mean instead of

```
1069         self.input.basis = Group({
1070             "eCut": "EnCut",
1071             "kPoint": Group({
1072                 "coords": "KpointCoords",
1073                 "weight": 1,
1074                 "relative": True
1075             }),
1076             "folding": "KpointFolding",
1077         })
```

You can put

```
1069         self.input.basis = Group({
1070             "eCut": "EnCut/13.606",
1071             "kPoint": Group({
1072                 "coords": "KpointCoords",
1073                 "weight": 1,
1074                 "relative": True
1075             }),
1076             "folding": "KpointFolding",
1077         })
```

Reply:
But maybe this is a question of taste. I prefer not to have exceptions in `to_sx_str` and I also think it's rather rare that the value is changed via `job.input.basis`. Besides I guess if the user sees `EnCut/13.606` it is clear that there must be a change of unit. On the other hand some people might find it confusing to see different units for `EnCut` and `eCut`.

Reply:
I agree; I removed the exception from `to_sx_str` and built the conversion into the default `job.input.basis` (line 1070)